packages/lesswrong/lib/collections/userTagRels/helpers.ts

```diff
@@ -0,0 +1,20 @@
+import { createMutator, updateMutator } from "../../../server/vulcan-lib/mutators";
+import UserTagRels from "./collection";
+
+export async function recordSubforumView(userId: string, tagId: string) {
+  const existingRel = await UserTagRels.findOne({userId, tagId});
+  if (existingRel) {
```

Comment:
I did it as an upsert initially but changed it because:
 - doing an upsert results in the `_id` being an ObjectId rather than a string, I wasn't sure if this was bad
 - I wanted to use `updateMutator` because it runs the callbacks and there is no way to pass the upsert option in to it (it would be easy to add this option but I would expect it to break some of the "before" callbacks)